protocol/contracts/templegold/TempleGoldStaking.sol

```diff
@@ -93,17 +77,15 @@ contract TempleGoldStaking is ITempleGoldStaking, TempleElevatedAccess, Pausable
         stakingToken = IERC20(_stakingToken);
         rewardToken = IERC20(_rewardToken);
     }
-    
+
     /**
-     * @notice Set vesting period for stakers
-     * @param _period Vesting period
+     * @notice Set unstake cooldown
+     * @param _period Cooldown period
      */
-    function setVestingPeriod(uint32 _period) external override onlyElevatedAccess {
-        if (_period < WEEK_LENGTH) { revert CommonEventsAndErrors.InvalidParam(); }
-        // only change after reward epoch ends
-        if (rewardData.periodFinish >= block.timestamp) { revert InvalidOperation(); }
-        vestingPeriod = _period;
-        emit VestingPeriodSet(_period);
+    function setUnstakeCooldown(uint32 _period) external override onlyElevatedAccess {
+        if (_period== 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
```

Comment:
Is this check required? If set to zero it disables the check which may be a feature?

Reply:
agreed